File: src/dataset/c/bob_exercism-IrvinRamiA/claude/ClaudeSonnet4_bob_v4.c
#include "bob.h"
#include <stdlib.h>
#include <ctype.h>
#include <stdbool.h>
#include <string.h>

char *hey_bob(char *greeting)
{
    if (!greeting) {
        char *response = malloc(22);
        strcpy(response, "Fine. Be that way!");
        return response;
    }

    bool has_letters = false;
    bool has_lowercase = false;
    bool is_question = false;
    bool has_content = false;
    
    char *ptr = greeting;
    char *last_non_space = NULL;
    
    while (*ptr) {
        if (!isspace(*ptr)) {
            has_content = true;
            last_non_space = ptr;
        }
        
        if (isalpha(*ptr)) {
            has_letters = true;
            if (islower(*ptr)) {
                has_lowercase = true;
            }
        }
        ptr++;
    }
    
    if (last_non_space && *last_non_space == '?') {
        is_question = true;
    }
    
    bool is_yelling = has_letters && !has_lowercase;
    
    char *response;
    
    if (!has_content) {
        response = malloc(22);
        strcpy(response, "Fine. Be that way!");
    }
    else if (is_question && is_yelling) {
        response = malloc(36);
        strcpy(response, "Calm down, I know what I'm doing!");
    }
    else if (is_question) {
        response = malloc(6);
        strcpy(response, "Sure.");
    }
    else if (is_yelling) {
        response = malloc(18);
        strcpy(response, "Whoa, chill out!");
    }
    else {
        response = malloc(10);
        strcpy(response, "Whatever.");
    }
    
    return response;
}

File: src/dataset/javascript/bob_exercism-ThomasZumsteg/claude/ClaudeSonnet4_bob_v4.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
	var trimmed = input.trim();
	
	if (!trimmed) {
		return "Fine. Be that way!";
	}
	
	var isQuestion = input.charCodeAt(input.length - 1) === 63;
	var hasLetters = false;
	var isAllUpper = true;
	
	for (var i = 0; i < input.length; i++) {
		var code = input.charCodeAt(i);
		if (code >= 65 && code <= 90) {
			hasLetters = true;
		} else if (code >= 97 && code <= 122) {
			hasLetters = true;
			isAllUpper = false;
		}
	}
	
	if (hasLetters && isAllUpper) {
		return "Whoa, chill out!";
	}
	
	if (isQuestion) {
		return "Sure.";
	}
	
	return "Whatever.";
};

module.exports = Bob;